backend/src/tests/neighborhood_api.test.ts

```diff
@@ -98,20 +103,87 @@ describe('When neighborhoods already exist in the db', () => {
   });
 
   test('GET /neighborhoods/id existing id returns single neignborhood', async () => {
+    const loginResponse = await loginUser(BOBS_LOGIN_DATA);
+    const { token } = loginResponse.body;
+
     const neighborhood = await prismaClient.neighborhood.findFirst({
       where: { name: "Bob's Neighborhood" },
+      include: {
+        admin: true,
+        users: true,
+        requests: true,
+      },
     });
     const id = neighborhood?.id;
-    const response = await api.get(`/api/neighborhoods/${id}`);
+    const response = await api.get(`/api/neighborhoods/${id}`).set('Authorization', `Bearer ${token}`);
     expect(response.status).toEqual(200);
     expect(response.body.id).toEqual(id);
+    expect(response.body).toHaveProperty('admin');
+    expect(response.body).toHaveProperty('users');
+    expect(response.body).toHaveProperty('requests');
   });
 
   test('GET /neighborhoods/id invalid id returns expected error', async () => {
-    const response = await api.get(`/api/neighborhoods/0`);
+    const loginResponse = await loginUser(BOBS_LOGIN_DATA);
+    const { token } = loginResponse.body;
+
+    const response = await api.get('/api/neighborhoods/0').set('Authorization', `Bearer ${token}`);
     expect(response.status).toEqual(404);
     expect(response.body.error).toEqual('No Neighborhood found');
   });
+
+  test('GET /neighborhoods/id only returns the id, name, description and location of neighborhood if user is not logged in', async () => {
+    const neighborhood = await prismaClient.neighborhood.findFirst({
+      where: { name: "Bob's Neighborhood" },
+      select: {
+        id: true,
+        name: true,
+        description: true,
+        location: true,
+      },
+    });
+    const id = neighborhood?.id;
+    const response = await api.get(`/api/neighborhoods/${id}`);
+
+    const neighborhoodFromDBKeys = Object.keys(neighborhood as object);
```

Comment:
Can't we use `as Neighborhood` here instead of `object`?

Reply:
Yes, you're right, will change it.